src/masternode-payments.cpp

```
@@ -273,12 +285,49 @@ bool CMasternodePayments::IsTransactionValid(const CTransaction& txNew, int nBlo
+        serialContext = spark::getSerialContext(txNew);
+    }
+
     for (const auto& txout : voutMasternodePayments) {
         bool found = false;
-        for (const auto& txout2 : txNew.vout) {
-            if (txout == txout2) {
-                found = true;
-                break;
+        spark::Address addr(params);
+        if (spark::IsPayToSparkAddress(txout.scriptPubKey, addr) && !coins.empty()) {
+            for (const auto  &coin: coins) {
```

cppcheck

[performance] Prefer prefix ++/-- operators for non-primitive types.

(postfixOperator)